src/app/modules/info/info-button/info-button.component.spec.ts

```diff
@@ -28,4 +28,11 @@ describe('InfoButtonComponent', () => {
     find('mat-icon').triggerEventHandler('click', '');
     expect(spy).toHaveBeenCalled();
   });
+
+  it('launchInfoDialog opens dialog box', async () => {
+    const { find, instance } = await shallow.mock(MatDialog, mockMatDialog).render();
+    instance.launchInfoDialog();
+    const nativeElement = find('.mat dialog container');
+    expect(nativeElement).toBeDefined();
```

Comment:
This does not test what you are expecting it to. To test whether it has any matching elements assert that  `nativeElement.length` is not 0. (Also poor naming. `nativeElement` is not the actual native element! It is the debug element)

Reply:
Would  `expect(find('.mat dialog container')).toHaveFoundOne();`work?

Reply:
No, since we're mocking MatDialog calling open will actually not open anything, rather you should test that the launchInfoDialog method tries to open a dialog by asserting that the open method has been called.

Reply:
No, since we're mocking MatDialog calling open will actually not open anything, rather you should test that the launchInfoDialog method tries to open a dialog by asserting that the open method has been called.

Reply:
I'm trying to spy on the open method but it's not letting me. What am I missing here? 

`   const { instance } = await shallow.mock(MatDialog, mockMatDialog).render();
    const spy = spyOn(mockMatDialog, 'open');
    instance.launchInfoDialog();
    expect(spy).toHaveBeenCalled();`